scrapy/core/downloader/handlers/http11.py

```diff
@@ -230,18 +239,20 @@ class TunnelingAgent(Agent):
 
     def __init__(
         self,
-        reactor,
-        proxyConf,
-        contextFactory=None,
-        connectTimeout=None,
-        bindAddress=None,
-        pool=None,
+        reactor: ReactorBase,
+        proxyConf: Tuple[str, int, Optional[bytes]],
+        contextFactory: Optional[IPolicyForHTTPS] = None,
+        connectTimeout: Optional[float] = None,
+        bindAddress: Optional[bytes] = None,
+        pool: Optional[HTTPConnectionPool] = None,
     ):
+        # TODO make this arg required instead
+        assert contextFactory is not None
```

Comment:
To do in this PR, or should we create an issue to track this?

Reply:
Not here, I'll make an issue for both instances. Not sure how to properly do this though. We could make all args kw-only.

Reply:
https://github.com/scrapy/scrapy/issues/6343